### PR TITLE
Add "output_type" parameter to major functions

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
_modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_

# Description

This PR serves as a solution to issue #306 and allows for a user that uses the major functions to specify the output file type instead of only being able to output files as CSV or CSV compressed. A user would be able to utilize the `output` functions ability to output files as `parquet`. 

## What is the nature of your change?

- [X] Bug fix (fixes an issue).
- [X] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

**Comment by gwaybio** [comment]
@jenna-tomkinson - thanks for this PR! Can you add tests to each function to confirm that the files they write are either csv or parquet based on this function argument?

**Comment by jenna-tomkinson** [comment]
@gwaybio I just added tests that I believe work/pass for the output function in https://github.com/cytomining/pycytominer/pull/307/commits/20d5a35c7202fa26ee981a4bbb89d4e86bffb4c6.

Feel free to finish your review now! 😄

File: pycytominer/tests/test_aggregate.py [hunk 1]
```diff
@@ -10,7 +10,8 @@
 tmpdir = tempfile.gettempdir()
 
 # Setup a testing file
-test_output_file = os.path.join(tmpdir, "test.csv")
+test_output_file1 = os.path.join(tmpdir, "test.csv")
+test_output_file2 = os.path.join(tmpdir, "test.parquet")
```
Comment (gwaybio) [inline pycytominer/tests/test_aggregate.py:14]:
```suggestion
OUTPUT_FILE_CSV = os.path.join(tmpdir, "test.csv")
OUTPUT_FILE_PARQUET = os.path.join(tmpdir, "test.parquet")
```

More specific than 1 and 2, and easier to parse.

File: pycytominer/tests/test_aggregate.py [hunk 2]
```diff
@@ -307,3 +308,32 @@ def test_custom_objectnumber_feature():
...
+            output_type=_type,
+        )
+
+    # read files in with pandas
+    csv_df = pd.read_csv(test_output_file1)
+    parquet_df = pd.read_parquet(test_output_file2)
+
+    # check to make sure the files were read in corrrectly as a pd.Dataframe
+    assert type(csv_df) == pd.DataFrame
+    assert type(parquet_df) == pd.DataFrame
```
Comment (gwaybio) [inline pycytominer/tests/test_aggregate.py:339]: can you also assert their equivalence? 

I think the function is `pd.testing.assert_frame_equal(csc_df, parquet_df)`

File: pycytominer/tests/test_aggregate.py (hunk 1 above)
Comment (gwaybio) [reply pycytominer/tests/test_aggregate.py:14]: Also note the capitalization to keep this test consistent with the other `test_.py` files.

File: pycytominer/tests/test_aggregate.py (hunk 2 above)
Comment (jenna-tomkinson) [reply pycytominer/tests/test_aggregate.py:339]: I have added it now!

**Comment by jenna-tomkinson** [comment]
@gwaybio Ready for review again!

**Review by gwaybio — APPROVED** [review]
LGTM, I will merge after passes tests

**Pull request merged**